boards/arm/blackpill_f411ce/blackpill_f411ce.dts

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright (c) 2020 Brian Bradley <brian.bradley.p@gmail.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/dts-v1/;
+#include <st/f4/stm32f411Xe.dtsi>
+
+/ {
+	model = "WeAct Studio Black Pill V2.0 Board";
+	compatible = "weact,blackpill-f411ce","st,stm32f411";
```

Comment:
Remove the SoC compatibles from this line, they are not needed in Board DTS

Reply:
@ioannisg currently lot of boards have this compatible. For coherency, should they be all cleaned-up?
If this is not the case, I'd say it has to be kept

Reply:
I did it this way in accordance with how other boards do it. To be honest I am not 100% sure why it is done this way, but I was trying to be consistent. I did consult with the documentation for this part, and believed this was in alignment with this recommendation.
``` 
The compatible property can have multiple values. Additional values are useful when the device is a specific instance of a more general family, to allow the system to match from most- to least-specific device drivers.
```
I am not sure if the way I have it here accomplishes that in any functional way, but this was part of my reasoning (but primarily because most other boards do this). 

I am happy to change it, but I just want to make sure I understand the generally agreed upon method.

Reply:
I think this is because of an incomplete transition to a specific policy.

For Nordic boards the node path `/soc/compatible` in `dts/arm/nordic/*` provides the SoC compatible list.  This makes some sense, so when you're looking for an SoC compatible you look in `/soc/compatible` rather than `/compatible` where the root identifies the complete target (board).

ST has not moved to this (undocumented?) practice.  Until it's agreed this is the way things should be done, I think boards should continue to store the SoC compatible in the board compatibles list, so it's associated with the board.  It can be removed once the SoC dtsi files have been updated.

@galak what do you think?

Reply:
@pabigot, this makes sense and we can move ST targets to comply with this if there is a general (and documented) agreement on that.